core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -1087,8 +1087,10 @@ protected function addMessage($strMessage, $strType)
 	 * @deprecated Deprecated since Contao 4.0, to be removed in Contao 5.0.
 	 *             Use Message::generate() instead.
 	 */
-	protected function getMessages($strScope=TL_MODE)
+	protected function getMessages($strScope=null)
 	{
+		$strScope = $strScope ?? TL_MODE;
```

Comment:
```suggestion
        // Set default scope in method body (see #3495)
		$strScope = $strScope ?? TL_MODE;
```

Reply:
I relly like the comments in Contao's code base pointing to issues with background information on why some specific logic is required.

However, is this comment really helpful if you are not aware of the context that TL_MODE was previously set as the default parameter in the method?

Reply:
I think it would be helpful to know for future reference (or deprecation) why we set a default in the method body here, as we don't usually do that (I think?).